activesupport/lib/active_support/cache.rb

```diff
@@ -815,23 +808,24 @@ def dup_value!
       end
 
       private
-        def compress!(compress_threshold)
+        def calculate_size_and_possibly_compress!(compress, compress_threshold)
           case @value
           when nil, true, false, Numeric
```

Comment:
Before entries with values as `true`, `false` and numeric had sized different of `0`, now they are also `0`.

```ruby
>> ActiveSupport::Cache::Entry.new(true).size
=> 3
>> ActiveSupport::Cache::Entry.new(false).size
=> 3
>> ActiveSupport::Cache::Entry.new(1_000_000_000).size
=> 8
```

We should still not compress then (this is probably why we were setting `0` to the `uncompressed_size`, but we should set the `@value_size` to the result of  `Marshal.dump(@value).bytesize`.